pkg/storage/legacysql/dualwrite/dualwriter.go

```diff
@@ -105,12 +105,13 @@ func (d *dualWriter) Create(ctx context.Context, in runtime.Object, createValida
 		if err != nil {
 			log.Error("unable to cleanup object in legacy storage", "err", err)
 		}
+		return storageObj, errObjectSt
 	}
 
 	if d.readUnified {
 		return storageObj, errObjectSt
```

Comment:
Suggestion: (Not related to this change) I just see it now, but that's kinda confusing. `errObjectSt` will always be nil, so we might just want to return nil here to indicate the success path.